### DISCUSSION TITLE
Display Env value with Project name

### DISCUSSION THREAD

**Discussion by sandyksit** [directus/directus#13129] [Archive]
Hi Team,
@rijkvanzanten is there any configuration for env. variable (like dev, test, prod), and I need to display left corner with 'Project Name' like 'Test project ('DEV env)', how we can display this with dynamic configuration.

Thanks,
Sandeep

**Comment by azrikahar** [comment]
I believe the Project Descriptor field in Project Settings should fit your description here:

https://user-images.githubusercontent.com/42867097/166895045-0d9a47c6-712a-41ca-b8cb-01f15251d2c9.mp4

Although you can't exactly modify it dynamically using an env. variable as you mentioned, aside from changing it directly in the App, you can either [update it via the API](https://docs.directus.io/reference/system/settings/) or use a [custom migration](https://docs.directus.io/extensions/migrations/#custom-migrations) depending on what works best for you.

**Reply by sandyksit** [reply]
@azrikahar I need to display ENV value on load of the project, Could you please guide me how we can call the api on load?
Thanks,
Sandeep

**Reply by darioguarascio** [reply]
@sandyksit What you can do is to run an API call after directus has started, to update the value of that field:

```
PATCH /settings
{
	"project_descriptor": $YOUR_VAR
}
```

**Reply by sandyksit** [reply]
@azrikahar @darioguarascio we have extension folder, where we can exactly call this api? I have called other api's from interface module. but for this i am not able to get where it will call.

**Reply by darioguarascio** [reply]
@sandyksit I was thinking more of a script that as step 1 starts up directus and step 2 updates the value of settings with a `curl` command

**Reply by sandyksit** [reply]
@azrikahar @darioguarascio from curl command we don't want to execute with token and credential, Is there any possibility call from extensions folder on load the application?

**Reply by azrikahar** [reply]
Perhaps you can try using an `app.after` Init hook with SettingsService to update the project_desriptor every time your Directus instance starts.🙂

Here's an example:

```js
export default ({ init }, { services, getSchema }) => {
	const { SettingsService } = services;

	init('app.after', async () => {
		const schema = await getSchema();
		const settingsService = new SettingsService({ schema });
		await settingsService.upsertSingleton({ project_descriptor: 'My Project' });
	});
};
```

**Reply by sandyksit** [reply]
Thanks @azrikahar , you guys are awesome. this is working as per expected.
But I think we have another options:

**Config Directus w a user/role that can be used to update the project_descriptor**

- Set up a special role "SystemSettingsRole" - which would only be used for updating this value. This role has a rule which only allows access to the "directus_settings" collection. and only matches project_descriptor.

-  Set up a special user "SystemSettings User" - which would only be used for updating this value.

This user has a Access Token set up. This Access token would be used in the curl command below
**Update the project_descriptor via the API after Directus starts up**
This could be done in the bin/startContainer.sh script
check the environment – if dev or uat, update the project_descriptor.
